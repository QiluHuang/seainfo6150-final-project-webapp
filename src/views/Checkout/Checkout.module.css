.container {
  display: flex;
  flex-direction: column;
  margin-top: 5%;
  margin-left: 20%;
  margin-right: 20%;
}

.wrapper {
  margin: 10px;
}


.labelText {
  display: inline-block;
	text-align: justify;
	vertical-align: middle;
  width: 300px;
  font-size: 20px;
  font-weight: 700;
  color: #329735;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1em;
}

.check {
  border-style: solid;
  /* border: 10px; */
  border-color: #329735;
  border-radius: 15px;
  box-shadow: 5px 5px #888888;
  margin: 20px;
}

.sentence {
  font-size: 20px;
  font-weight: 700;
  color: black;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1em;
  text-decoration: underline;
}

.buttonWrapper {
  margin-top: 20px;
}

.button {
    background-color: #329735;
    border: none;
    color: white;
    padding: 16px 32px;
    text-align: center;
    font-size: 16px;
    margin: 4px 2px;
    opacity: 0.6;
    transition: 0.3s;
    display: inline-block;
    text-decoration: none;
    cursor: pointer;
  }
  
  .button:hover {
    opacity: 1
  }