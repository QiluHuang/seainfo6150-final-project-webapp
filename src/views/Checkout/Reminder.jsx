import React from 'react';
import styles from './Reminder.module.css';
import PropTypes from "prop-types";


const Reminder = (props) => {
    return (
        <div className={styles.container}>
            <div className={styles.image}><img src="https://cdn11.bigcommerce.com/s-7j8u58/images/stencil/1110x1110/uploaded_images/miami-healthy-meals-delivery-service-the-best-way-to-time-your-meals-for-weight-loss.jpg?t=1561061586" alt={"background"}/></div>

            <div className={styles.background}>
                <div className={styles.text}>Please get your meal at {props.text}</div>
            </div>
        </div>
    );
}

Reminder.propTypes = {
    text: PropTypes.string.isRequired,
};

export default Reminder;