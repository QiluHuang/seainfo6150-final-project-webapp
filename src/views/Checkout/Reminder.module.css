.container {
    background-image: url("https://cdn11.bigcommerce.com/s-7j8u58/images/stencil/1110x1110/uploaded_images/miami-healthy-meals-delivery-service-the-best-way-to-time-your-meals-for-weight-loss.jpg?t=1561061586");
    /* background-repeat: no-repeat; */
    /* background-attachment: fixed; */
    background-size: cover;
}

.image {
    width: 100%;
    /* max-height: 100%; */
    height: auto;
    visibility: hidden;
}

.background {
    max-width: 100%;
    /* height: 1000px; */
}

.text {
    position: absolute;
    top: 100%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: "Sofia", sans-serif;
    color: white;
    font-style: bold;
    font-size: 3em;
    text-shadow: 5px 5px #5c5c5c;
}