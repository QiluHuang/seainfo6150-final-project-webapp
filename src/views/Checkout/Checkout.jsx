import React, { useState } from "react";
import Button from "../../components/Buttons/Button/Button";
import styles from './Checkout.module.css';
import Reminder from "./Reminder";

const Checkout = () => {
  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };


  return (
    <div>
      {
        submittedForm ? (
          <Reminder text={submittedForm.get("myDropdown")}/>
          // <div>Your form entry was {submittedForm.get("myText")} and {submittedForm.get("myDropdown")}</div>
        ) : (
          <form onSubmit={onSubmit}>
            <div className={styles.container}>
              <div className={styles.wrapper}>
                  <label className={styles.labelText}> Name: </label>
                  <input type="text" name="customerName" id="customerNameId" placeholder="John Smith" maxLength="255" size="50"/>
              </div>

              <div className={styles.wrapper}>
                  <label className={styles.labelText}>Tel: </label>
                  <input type="text" name="tel" id="telId" maxLength="255" size="50" placeholder="123-456-7890" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" required/>
              </div>

              <div className={styles.wrapper}>
                  <label className={styles.labelText}>Email: </label>
                  <input type="email" name="tel" id="telId" maxLength="255" size="50" placeholder="john_smith@gmail.com" required/>
              </div>

              <div className={styles.wrapper}>     
                <div className={styles.select_box}>
                    <label className={styles.labelText}>Time for meal </label>
                    <select name="myDropdown" id="myDropdownId">
                    <option value="11:30AM">11:30AM</option>
                    <option value="12:30PM">12:30PM</option>
                    <option value="01:30PM">01:30PM</option>
                    <option value="05:30PM">05:30PM</option>
                    <option value="06:30PM">06:30PM</option>
                    <option value="07:30PM">07:30PM</option>
                    <option value="08:30PM">08:30PM</option>
                    <option value="09:30PM">09:30PM</option>
                    </select>
                </div>
              </div>

              <div className={styles.wrapper}>
                <div className={styles.selectbox}>
                    <label className={styles.labelText}>Way to get your meal </label>
                    <select name="myDropdown" id="myDropdownId">
                    <option value="Delivery">Delivery</option>
                    <option value="Drive Thru">Drive Thru</option>
                    </select>
                </div>
              </div>

              <div className={styles.check}>
                <ul>
                    <div className={styles.sentence}>We are care about your health and each meal. Please check the following information.</div>
                    <li>
                      <input type="checkbox" id="check1" name="membership" value="true" />
                      <label> I am membership. </label>
                      <input type="text" id="membershipInfo" placeholder="please enter your membership number." maxLength="255" size="50" pattern="[0-9]{5}"required/>
                    </li>
                    <li>
                      <input type="checkbox" id="check2" name="spicy" value="true" />
                      <label> I can eat spicy food. </label>
                    </li>
                    <li>
                      <input type="checkbox" id="check3" name="allergy" value="true" />
                      <label> I am allergic to certain ingredients. </label>
                      <input type="text" name="allergic_ingredients" id="ingredientID" maxLength="255" size="50"/>

                    </li>
                </ul>
              </div>


              <div className={styles.buttonWrapper}>
                  <input className={styles.button} type="submit" value="Submit" />
              </div>
            </div>
          </form>
        )
      }
    </div>
  )
}

export default Checkout;
