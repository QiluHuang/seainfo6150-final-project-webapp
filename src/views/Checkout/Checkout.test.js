import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import Checkout from "./Checkout.jsx";

describe("Checkout tests", () => {
    it("renders the Checkout component correctly", () => {

    const { container } = render(<Router><Checkout /></Router>);
    expect(container).toMatchSnapshot();
    });
});