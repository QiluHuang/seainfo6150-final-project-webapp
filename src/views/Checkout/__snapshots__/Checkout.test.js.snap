// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout tests renders the Checkout component correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="container"
      >
        <div
          class="wrapper"
        >
          <label
            class="labelText"
          >
             Name: 
          </label>
          <input
            id="customerNameId"
            maxlength="255"
            name="customerName"
            placeholder="John Smith"
            size="50"
            type="text"
          />
        </div>
        <div
          class="wrapper"
        >
          <label
            class="labelText"
          >
            Tel: 
          </label>
          <input
            id="telId"
            maxlength="255"
            name="tel"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            placeholder="123-456-7890"
            required=""
            size="50"
            type="text"
          />
        </div>
        <div
          class="wrapper"
        >
          <label
            class="labelText"
          >
            Email: 
          </label>
          <input
            id="telId"
            maxlength="255"
            name="tel"
            placeholder="john_smith@gmail.com"
            required=""
            size="50"
            type="email"
          />
        </div>
        <div
          class="wrapper"
        >
          <div
            class="select_box"
          >
            <label
              class="labelText"
            >
              Time for meal 
            </label>
            <select
              id="myDropdownId"
              name="myDropdown"
            >
              <option
                value="11:30AM"
              >
                11:30AM
              </option>
              <option
                value="12:30PM"
              >
                12:30PM
              </option>
              <option
                value="01:30PM"
              >
                01:30PM
              </option>
              <option
                value="05:30PM"
              >
                05:30PM
              </option>
              <option
                value="06:30PM"
              >
                06:30PM
              </option>
              <option
                value="07:30PM"
              >
                07:30PM
              </option>
              <option
                value="08:30PM"
              >
                08:30PM
              </option>
              <option
                value="09:30PM"
              >
                09:30PM
              </option>
            </select>
          </div>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="selectbox"
          >
            <label
              class="labelText"
            >
              Way to get your meal 
            </label>
            <select
              id="myDropdownId"
              name="myDropdown"
            >
              <option
                value="Delivery"
              >
                Delivery
              </option>
              <option
                value="Drive Thru"
              >
                Drive Thru
              </option>
            </select>
          </div>
        </div>
        <div
          class="check"
        >
          <ul>
            <div
              class="sentence"
            >
              We are care about your health and each meal. Please check the following information.
            </div>
            <li>
              <input
                id="check1"
                name="membership"
                type="checkbox"
                value="true"
              />
              <label>
                 I am membership. 
              </label>
              <input
                id="membershipInfo"
                maxlength="255"
                pattern="[0-9]{5}"
                placeholder="please enter your membership number."
                required=""
                size="50"
                type="text"
              />
            </li>
            <li>
              <input
                id="check2"
                name="spicy"
                type="checkbox"
                value="true"
              />
              <label>
                 I can eat spicy food. 
              </label>
            </li>
            <li>
              <input
                id="check3"
                name="allergy"
                type="checkbox"
                value="true"
              />
              <label>
                 I am allergic to certain ingredients. 
              </label>
              <input
                id="ingredientID"
                maxlength="255"
                name="allergic_ingredients"
                size="50"
                type="text"
              />
            </li>
          </ul>
        </div>
        <div
          class="buttonWrapper"
        >
          <input
            class="button"
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
